project/Build.scala

```diff
@@ -1216,8 +1226,50 @@ object Build extends sbt.Build {
     base = file("test-suite/js"),
     settings = commonSettings ++ myScalaJSSettings ++ testTagSettings ++
       testSuiteCommonSettings(isJSTest = true) ++ Seq(
-        name := "Scala.js test suite",
+        name := "Scala.js test suite"
+      ) ++ inConfig(Test)(Seq(
+        artifactPath in jsTestDefinitions :=
+          crossTarget.value / "test_definitions.js",
+
+        // Generate a JS file that contains defined tests and testFrameworks.
+        // The file will be called "test_definitions.js" and defines:
+        // - var definedTests
+        // - var testFrameworkNames
+        jsTestDefinitions := {
+          import org.scalajs.testadapter.TaskDefSerializers._
+
+          val outFile = (artifactPath in jsTestDefinitions).value
+
+          val allTests = for {
+            t <- definedTests.value.toList
+          } yield {
+            new sbt.testing.TaskDef(t.name, t.fingerprint,
+                t.explicitlySpecified, t.selectors)
+          }
+
+          val allFrameworks =
+            loadedTestFrameworks.value.map(_._1.implClassNames.toList)
+
+          val writer = new BufferedWriter(new FileWriter(outFile))
+
+          try {
+            writer.write("var definedTests = ")
+            writeJSON(allTests.toJSON, writer)
+            writer.write(";");
+
+            writer.write("var testFrameworkNames = ")
+            writeJSON(allFrameworks.toList.toJSON, writer)
+            writer.write(";")
+          } finally {
+            writer.close()
+          }
+
+          outFile
+        },
+        jsTestDefinitions <<= jsTestDefinitions triggeredBy definedTests,
```

Comment:
Otherwise the "test_definitions.js" wont be generated for the `testSuite` project.